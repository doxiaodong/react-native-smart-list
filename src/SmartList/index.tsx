import React, { Component } from 'react'
import { View } from 'react-native'

export default class SmartList extends Component<{
  tabs: React.ReactElement<any>
}, {}> {
  render() {
    const { tabs, children } = this.props
    return (
      <View>
        <View>
          {tabs}
        </View>
        <View>
          {children}
        </View>
      </View>
    )
  }
}
