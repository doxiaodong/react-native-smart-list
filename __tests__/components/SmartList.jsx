import React from 'react'
import { View, Text } from 'react-native'
import renderer from 'react-test-renderer'
import SmartList, { Tabs, Pane } from '@/index'

describe('SmartList Component', () => {
  test('Snapshot', () => {
    const component = renderer.create(
      <SmartList
        tabs={
          <Tabs>
            <Pane label="1">111</Pane>
            <Pane label="2">222</Pane>
            <Pane label="3">333</Pane>
          </Tabs>
        }
      >
        <View>
          <Text>List</Text>
        </View>
      </SmartList>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
