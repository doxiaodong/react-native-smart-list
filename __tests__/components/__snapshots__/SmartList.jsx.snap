// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmartList Component Snapshot 1`] = `
<View>
  <View>
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
          >
            1
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
          >
            2
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
          >
            3
          </Text>
        </View>
      </View>
      <View>
        <View />
        <View />
        <View />
      </View>
    </View>
  </View>
  <View>
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
      >
        List
      </Text>
    </View>
  </View>
</View>
`;
